include/h2/core/sync.hpp

```diff
@@ -239,6 +239,39 @@ class SyncEvent
     : device(Device::GPU), gpu_event(raw_event) {}
 #endif
 
+  SyncEvent(const SyncEvent&) = default;
+  SyncEvent& operator=(const SyncEvent&) = default;
+
+  SyncEvent(SyncEvent&& other)
+    : device(other.device)
+  {
+    H2_DEVICE_DISPATCH(
+        device,
+        {
+          cpu_event = other.cpu_event;
+          other.cpu_event = 0;
+        },
+        {
+          gpu_event = other.gpu_event;
+          other.gpu_event = nullptr;
+        });
+  }
+  SyncEvent& operator=(SyncEvent&& other)
+  {
+    device = other.device;
+    H2_DEVICE_DISPATCH(
+        device,
+        {
+          cpu_event = other.cpu_event;
+          other.cpu_event = 0;
+        },
+        {
+          gpu_event = other.gpu_event;
+          other.gpu_event = nullptr;
```

Comment:
```suggestion
          gpu_event = std::exchange(other.gpu_event, nullptr);
```
Handle assign-to-self.